test/componentDiagram-test.js

```diff
@@ -22,10 +22,37 @@ global.location = window.location;
 SVGElement.prototype.getBBox = () => ({x: 0, y: 0, width: 0, height: 0});
 SVGElement.prototype.getScreenCTM = () => (new DOMMatrix());
 
-test('should render a diagram', function (t) {
-  const model = readJSON('fullModel.json');
-  const elem = document.querySelector('#diagram');
-  const diagram = new cdiagram(elem);
-  diagram.render(model);
-  t.end();
+const model = readJSON('fullModel.json');
+const elem = document.querySelector('#diagram');
```

Comment:
Does every test create a new DOM globally?

Reply:
No, DOM is created once and all tests use the same diagram defined on line 15:
https://github.com/applandinc/d3-appmap/pull/31/commits/1b54761042d80c4aeb001a29fde606dfedcdeaa1#diff-5a4a775864b3028a3aac4dc0a7d97539f903f173f9cff5b7b1df816a99f2f478L15

Reply:
Isn't it a problem to re-use the DOM between tests? Tests should be independent of each other so that the state of one test does not influence the state of another.

Reply:
My idea was in creating one diagram in the DOM and run multiple tests that will use the same instance of the diagram.
The same thing is happening in `rspec` tests when we load the page and work with that page in headless Chrome

Reply:
It's OK to make multiple assertions within a test, but each test needs to use its own fresh diagram and DOM. The `rspec` tests actually create a new browser window and new session for each test.

Reply:
Okay, got it.
I'll make every test to use it's own DOM.